stor/dx.py

```diff
@@ -455,29 +837,34 @@ def canonical_resource(self):
         Raises:
             DuplicateError: if filename is not unique
             NotFoundError: if resource is not found on DX platform
+            ValueError : if path is folder path
         """
         if not self.resource:
             return None
-        if self._is_folder():
-            raise ValueError('DXPath ({}) ending in folders cannot be canonicalized'.format(self))
+        if utils.has_trailing_slash(self):
+            raise ValueError('Invalid operation ({method}) on folder path ({path})'
+                             .format(path=self, method=sys._getframe(2).f_code.co_name))
         objects = [{
             'name': self.name,
             'folder': self.resource.parent,
-            'project': self.canonical_project
+            'project': self.canonical_project,
+            'batchsize': 2
         }]
         results = dxpy.resolve_data_objects(objects=objects)[0]
         if len(results) > 1:
-            raise DuplicateError('The virtual resource is not unique on DNAnexus')
+            raise DuplicateError('Multiple objects found at path ({}). '
+                                 'Try using a canonical ID instead'.format(self))
         elif len(results) == 1:
             return results[0]['id']
         else:
-            raise NotFoundError('The virtual resource does not exist on DNAnexus')
+            raise NotFoundError('No data object was found for the given path on DNAnexus')
```

Comment:
do you want to include the original path here like in OSError?

Reply:
Yep, included.